### ISSUE TITLE
I can't build CoreCLR against ICU 57.1

### ISSUE THREAD

**Issue by pkubaj** [dotnet/coreclr#7376]
I'm about to send a working FreeBSD port. It builds fine against ICU 55.1, but recent upgrade to 57.1 broke it.
It seems CoreCLR is not compatible with new ICU:
`[  0%] Building C object src/inc/CMakeFiles/corguids.dir/__/pal/prebuilt/idl/fusionpriv_i.c.o
/wrkdirs/usr/ports/devel/coreclr/work/coreclr-1.0.4/src/corefx/System.Globalization.Native/collation.cpp:282:9: error: 'ucol_setVariableTop' is deprecated [-Werror,-Wdeprecated-declarations]
        ucol_setVariableTop(pClonedCollator, ignoreSymbolsVariableTop, 1, pErr);
        ^
/usr/local/include/unicode/ucol.h:1254:1: note: 'ucol_setVariableTop' has been explicitly marked deprecated here
ucol_setVariableTop(UCollator *coll, 
^
`

**Comment by janvorli** [comment]
Seems similar to #5160 - maybe the detection of the ucol_setMaxVariable doesn't work properly with the latest ICU again.

**Comment by janvorli** [comment]
@pkubaj can you please check `CMakeFiles/CMakeError.log` in your build output folder (I guess it is in `bin/obj/FreeBSD.x64.Debug`)? Search for the `ucol_setMaxVariable`. That should show why it is not being detected.

**Comment by pkubaj** [comment]
In `CMakeFiles/CMakeError.log` I have:
```
Building C object CMakeFiles/cmTC_de990.dir/CheckSymbolExists.c.o
/usr/local/bin/clang37   -I/usr/local/include -I/usr/local/opt/icu4c/include  -Wall -std=c11    -o CMakeFiles/cmTC_de990.dir/CheckSymbolExists.c.o   -c /usr/local/poudriere/ports/default/devel/coreclr/work/coreclr-1.0.4/bin/obj/FreeBSD.x64.Debug/CMakeFiles/CMakeTmp/CheckSymbolExists.c
Linking C executable cmTC_de990
/usr/local/bin/cmake -E cmake_link_script CMakeFiles/cmTC_de990.dir/link.txt --verbose=1
/usr/local/bin/clang37   -Wall -std=c11     CMakeFiles/cmTC_de990.dir/CheckSymbolExists.c.o  -o cmTC_de990  
CMakeFiles/cmTC_de990.dir/CheckSymbolExists.c.o: In function `main':
/usr/local/poudriere/ports/default/devel/coreclr/work/coreclr-1.0.4/bin/obj/FreeBSD.x64.Debug/CMakeFiles/CMakeTmp/CheckSymbolExists.c:(.text+0x19): undefined reference to `ucol_setMaxVariable'
```
Although the necessary symbol is actually present:
pkubaj@dev:/usr/local/lib # readelf -a libicui18n.so | grep ucol_setMaxVariable
  6818: 0000000000142790    24 FUNC    GLOBAL DEFAULT   11 ucol_setMaxVariable

**Comment by wintersteiger** [comment]
I still see the same errors on FreeBSD with ICU 58.2:

```
#ifndef ucol_setMaxVariable
    return ((int*)(&ucol_setMaxVariable))[argc];
#...
```
throws
```
   ...  undefined reference to `ucol_setMaxVariable'
```
Plus the deprecation errors:
```
.../icushim.h:143:1: error: 'ucol_setVariableTop' is deprecated
```

**Comment by janvorli** [comment]
I have just found what's wrong. My change #7773 has disabled the find_library for ICU libraries in src/corefx/System.Globalization.Native/CMakeLists.txt when FEATURE_FIXED_ICU_VERSION is set, because the libraries are no longer needed for linking. However, I have not realized that they are still needed in the configure.cmake in the same folder to detect whether the ICU library contains ucol_setMaxVariable.

**Comment by janvorli** [comment]
@wintersteiger the #9990 will fix the issue once it is merged in. I've verified it on my local installation of FreeBSD 11 with the same version of ICU that you have.
@pkubaj I am sorry for the long response time, the issue somehow got out of my radar after I've asked you for the log.

**Comment by janvorli** [comment]
Fixed by #9990
